### PR TITLE
1.14 stuff

### PR DESCRIPTION
**Pull request by Cheaterpaul**

### CODE REVIEW

File: src/main/java/de/teamlapen/vampirism/client/render/particle/HealingParticle.java [hunk 1]
```diff
@@ -1,32 +1,18 @@
...
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 import javax.annotation.Nullable;
 
 public class HealingParticle extends SpriteTexturedParticle {
-    private final TextureManager textureManager = Minecraft.getInstance().getTextureManager();
-    private static final ResourceLocation HEART_TEXTURE = new ResourceLocation("vampirism", "textures/particle/particles.png");
-    private static final VertexFormat VERTEX_FORMAT = (new VertexFormat()).addElement(DefaultVertexFormats.POSITION_3F).addElement(DefaultVertexFormats.TEX_2F).addElement(DefaultVertexFormats.COLOR_4UB).addElement(DefaultVertexFormats.TEX_2S).addElement(DefaultVertexFormats.NORMAL_3B).addElement(DefaultVertexFormats.PADDING_1B);
+    //private static final VertexFormat VERTEX_FORMAT = (new VertexFormat()).addElement(DefaultVertexFormats.POSITION_3F).addElement(DefaultVertexFormats.TEX_2F).addElement(DefaultVertexFormats.COLOR_4UB).addElement(DefaultVertexFormats.TEX_2S).addElement(DefaultVertexFormats.NORMAL_3B).addElement(DefaultVertexFormats.PADDING_1B);
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/client/render/particle/HealingParticle.java:15]: Is this not needed anymore or does it have to be updated?

File: src/main/java/de/teamlapen/vampirism/core/ModEntities.java [hunk 2]
```diff
@@ -127,26 +135,25 @@ static void registerEntities(IForgeRegistry<EntityType<?>> registry) {
...
-        EntitySpawnPlacementRegistry.register(villager_hunter_faction, EntitySpawnPlacementRegistry.PlacementType.ON_GROUND, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
-        EntitySpawnPlacementRegistry.register(villager_vampire_faction, EntitySpawnPlacementRegistry.PlacementType.ON_GROUND, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
-        EntitySpawnPlacementRegistry.register(hunter_trainer_dummy, EntitySpawnPlacementRegistry.PlacementType.ON_GROUND, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
-        EntitySpawnPlacementRegistry.register(vampire, EntitySpawnPlacementRegistry.PlacementType.ON_GROUND, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
-        EntitySpawnPlacementRegistry.register(advanced_vampire, EntitySpawnPlacementRegistry.PlacementType.ON_GROUND, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
-        EntitySpawnPlacementRegistry.register(hunter_trainer, EntitySpawnPlacementRegistry.PlacementType.ON_GROUND, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
+//        EntitySpawnPlacementRegistry.register(blinding_bat, EntitySpawnPlacementRegistry.PlacementType.NO_RESTRICTIONS, Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, null);
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/core/ModEntities.java]: Same here?

File: src/main/java/de/teamlapen/vampirism/core/ModItems.java [hunk 3]
```diff
@@ -195,49 +192,30 @@ static void registerCraftingRecipes() {
...
-                ModItems.holy_water_bottle_normal, new ItemStack(ModItems.item_garlic, 4))
-                .setRequirements(1, HunterSkills.purified_garlic);
-        cauldronCraftingManager.addRecipe(new ItemStack(ModItems.garlic_beacon_core), ModItems.item_garlic, Blocks.BLACK_WOOL)
-                .setRequirements(1, HunterSkills.garlic_beacon);
-        cauldronCraftingManager
-                .addRecipe(ModItems.garlic_beacon_core_improved,
-                        ModItems.holy_water_bottle_ultimate, ModItems.garlic_beacon_core)
-                .setRequirements(1, HunterSkills.garlic_beacon_improved).setExperience(2F);
-        //cauldronCraftingManager.addRecipe(new ItemStack(ModItems.pure_salt, 4),new FluidStack(FluidRegistry.WATER, Fluid.BUCKET_VOLUME), null).setRequirements(1, HunterSkills.basic_alchemy).setCookingTime(20 * 60); TODO 1.14 fluid
+//        cauldronCraftingManager.registerLiquidColor(ModItems.holy_water_bottle_normal, 0x6666FF);
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/core/ModItems.java]: are cauldron recipes done?

File: src/main/java/de/teamlapen/vampirism/entity/vampire/BasicVampireEntity.java [hunk 4]
```diff
@@ -348,7 +346,7 @@ protected void registerGoals() {
         this.goalSelector.addGoal(1, new BreakDoorGoal(this, (difficulty) -> {
             return difficulty == net.minecraft.world.Difficulty.HARD;
         }));//Only break doors on hard difficulty
-        this.tasks_avoidHunter = new AvoidEntityGoal(this, CreatureEntity.class, 10, 1.0, 1.1, VampirismAPI.factionRegistry().getPredicate(getFaction(), false, true, false, false, VReference.HUNTER_FACTION));
+        this.tasks_avoidHunter = new AvoidEntityGoal<CreatureEntity>(this, CreatureEntity.class, 10, 1.0, 1.1, VampirismAPI.factionRegistry().getPredicate(getFaction(), false, true, false, false, VReference.HUNTER_FACTION));
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/entity/vampire/BasicVampireEntity.java:349]: Usually it is best to just use empty brackets <> for generics. Saves some typing when changing things and removes redundancy. Doesn't hurt though

File: src/main/java/de/teamlapen/vampirism/inventory/crafting/AlchemicalCauldronCraftingManager.java [hunk 5]
```diff
@@ -1,138 +1,136 @@
...
-        if (o instanceof ItemStack) return ((ItemStack) o).copy();
-        if (o instanceof Item) return new ItemStack((Item) o);
-        if (o instanceof Block) return new ItemStack((Block) o);
-        throw new IllegalArgumentException(TAG + ": Argument has to be one of the following: ItemStack, Item or Block");
-
-    }
-}
+//package de.teamlapen.vampirism.inventory.crafting;
+//
+//import com.google.common.collect.Lists;
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/inventory/crafting/AlchemicalCauldronCraftingManager.java]: If this has been completely replaced, feel free to delete it

File: src/main/java/de/teamlapen/vampirism/inventory/recipes/AlchemicalCauldronRecipe.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                ", experience=" + experience +
+                ", fluidStack=" + fluid +
+                '}';
+    }
+
+//    private void addFluidStackDescription(ItemStack stack, FluidStack fluidStack) {
+//
+//        CompoundNBT nbt = stack.getTag();
+//        if (nbt == null) nbt = new CompoundNBT();
+//        CompoundNBT display = nbt.contains("display", 10) ? nbt.getCompound("display") : new CompoundNBT();
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/inventory/recipes/AlchemicalCauldronRecipe.java]: Either add TODO or remove

File: src/main/java/de/teamlapen/vampirism/tileentity/AlchemicalCauldronTileEntity.java [hunk 7]
```diff
@@ -1,514 +1,282 @@
...
-    private ISoundReference boilingSound;
-    /**
-     * Can contain a recipe which can be cooked by the owner.
-     * Is set when a {@link IAlchemicalCauldronRecipe#canBeCooked(int, ISkillHandler)} check succeeds.
-     * As long as the current recipe is equals to this one, no new check is executed.
-     * Thereby the owner can e.g. disconnect and the cauldron can continue cooking (as long as the recipe does not change).
-     */
-    @Nullable
-    private IAlchemicalCauldronRecipe checkedRecipe;
+                if (this.isBurning() && this.func_214008_b(irecipe) && this.canPlayerCook(irecipe)) {
```
Comment (maxanier) [inline src/main/java/de/teamlapen/vampirism/tileentity/AlchemicalCauldronTileEntity.java]: When using SRG names feel free to add a comment saying what they might do, so once get mapped it is easier to identify what they do.
If this is just a plain copy of vanilla, add a reference instead

**Review by maxanier — COMMENTED** [review]
Awesome

File: src/main/java/de/teamlapen/vampirism/client/render/particle/HealingParticle.java (hunk 1 above)
Comment (Cheaterpaul) [reply src/main/java/de/teamlapen/vampirism/client/render/particle/HealingParticle.java:15]: i would test if it work with the vanilla particle ParticleTypes#HEART -> delete all files or if still needed write it (new)

File: src/main/java/de/teamlapen/vampirism/core/ModItems.java (hunk 3 above)
Comment (Cheaterpaul) [reply src/main/java/de/teamlapen/vampirism/core/ModItems.java]: except for the Fluid part yes

File: src/main/java/de/teamlapen/vampirism/player/vampire/actions/FreezeVampireAction.java [hunk 8]
```diff
@@ -27,7 +28,7 @@ public FreezeVampireAction() {
     @Override
     public boolean activate(final IVampirePlayer vampire) {
         PlayerEntity player = vampire.getRepresentingPlayer();
-        List<Entity> l = player.getEntityWorld().getEntitiesInAABBexcluding(player, player.getBoundingBox().grow(10, 5, 10), vampire.getNonFriendlySelector(true, false)::test);
+        List<Entity> l = player.getEntityWorld().getEntitiesInAABBexcluding(player, player.getBoundingBox().grow(10, 5, 10), vampire.getNonFriendlySelector(true, false));//TODO @Maxanier needs Predicate<Entity>
```
Comment (Cheaterpaul) [inline src/main/java/de/teamlapen/vampirism/player/vampire/actions/FreezeVampireAction.java:31]: @maxanier

File: src/main/java/de/teamlapen/vampirism/potion/blood/BloodPotionRegistry.java [hunk 9]
```diff
@@ -73,7 +74,7 @@ public IBloodPotionEffect getRandomEffect(@Nonnull ItemStack item, boolean bad,
         for (BloodPotionCategory category : categories) {
             if (!item.isEmpty() && category.containsItem(item)) {
                 for (WeightedEffect effect : category.getEffectsCopy()) {
-                    effect.itemWeight *= 5;
+                    effect.itemWeight *= 5; //TODO @Maxanier (AT public-f ?, doesnt work for me)
```
Comment (Cheaterpaul) [inline src/main/java/de/teamlapen/vampirism/potion/blood/BloodPotionRegistry.java:77]: @maxanier

File: src/main/java/de/teamlapen/vampirism/world/VampirismWorldData.java [hunk 10]
```diff
@@ -26,7 +27,7 @@
     VampirismWorldData get(@Nonnull World world) {
         String s = fileNameForProvider(world.dimension);
         DimensionType key = world.getDimension().getType().isVanilla() ? DimensionType.OVERWORLD : world.getDimension().getType();
-        VampirismWorldData data = world.getSavedData(key, VampirismWorldData::new, s);
+        VampirismWorldData data = world.getSavedData(key, VampirismWorldData::new, s);//TODO @Maxanier
```
Comment (Cheaterpaul) [inline src/main/java/de/teamlapen/vampirism/world/VampirismWorldData.java:30]: @maxanier

**Pull request merged**